README.md

```diff
@@ -438,6 +438,34 @@ defines the following functions:
   elimination leading up to the results passed to this function.
 
 
+### `constexpr_wrapper`: function arguments as constant expressions
```

Comment:
This is below the "Benchmark support functions" (and on the same level).

Still, this feels a bit strange.

Is this a "helper" for benchmarking? A helper to implement other stuff? An internal implementation detail?

If it gets more to the later, I'd suggest moving this into a new `.md` file.
… and possibly moving it into a subnamespace of `vir::`?

Reply:
It's the same header level (`###`) as benchmarking support. So it's another "additional feature" - in addition to the Parallelism TS 2 `simd` implementation.
It is used internally and can be useful for interacting with the execution policy modifiers (in the future). `constexpr_wrapper` types also get passed to user-supplied callables.

Hmm, what name would a subnamespace get? `vir::tools::`? I consider the `vir` namespace a staging ground for `std` :wink:. `vir::constexpr_wrapper` is an implementation of a (proposed) C++26 feature called `std::constexpr_wrapper`.

Reply:
I added a bit more information to the README. Before I add another `.md` file I should rather switch to doxygen (or some other "real" API documentation tool).

Reply:
Yeah, if `vir::` is a staging are, you might also want to differentiate the stuff that is immediately intended for staging and "related" stuff?

So yes, `vir::tools` doesn't sound too problematic. It still says "Hey, this is still public API, just not the core of this thing".

For those things that are (proposed) things for `std::`, maybe something like `vir::backport` or `vir::stdx`? I don't know, what is common sense there?

Reply:
OK, different question: Why do we put things in namespaces:
1. To avoid name clashes (This doesn't need namespaces, only unique names. E.g. C libraries achieve this by prefixing everything in a library with a certain name.)
2. To control associated namespaces (ADL). This is a technical reason and probably the most important one.
3. To allow users to "import" a namespace or to define a shorthand.

For `vir::constexpr_wrapper` I don't see any need to avoid name clashes (1) or to allow a shorter / unqualified use of the feature (3). ADL is always a concern, but sub-namespaces typically make the situation worse, rather than simpler.

So I'd rather avoid putting stuff into `vir::tools` with seemingly arbitrary reasoning. Why isn't everything in `vir` a tool and inside this namespace. What makes it a tool vs a non-tool?

FWIW, I go with Titus Winters who's statements on namespace I interpret as avoiding nested namespace unless there's a good *technical* reason. (IIRC)